txtorcon/circuit.py

```diff
@@ -289,12 +289,23 @@ def build_timeout_circuit(tor_state, reactor, path, timeout, using_guards=False)
     CircuitBuildTimedOutError will be raised unless we
     receive a circuit build result within the `timeout` duration.
     """
+    timed_circuit = None
     d = tor_state.build_circuit(path, using_guards)
-    reactor.callLater(timeout, d.cancel)
 
-    def trap_cancel(f):
+    def get_circuit(c):
+        timed_circuit = c
+        return c
```

Comment:
I think what you intend here isn't supported by Python (i.e. you want to assign to the outer-scope `timed_circuit`, but this assignment makes a local-only assignment to `timed_circuit` so I don't believe you can ever hit the "`is not None`" clause in the errback...